src/main/java/org/craftercms/engine/store/s3/S3SiteListResolver.java

```diff
@@ -58,24 +60,29 @@ public void setSiteNameMacroName(String siteNameMacroName) {
 
     @Override
     public Collection<String> getSiteList() {
-        String bucketName = s3Uri.getBucket();
+        String bucketName = s3Uri.bucket().orElseThrow();
```

Comment:
Let's specify the bucket name in the exception thrown message

Reply:
It only throws when the bucket is null so we won't have the bucket name:

    /**
     * Returns the bucket specified in the URI. Returns an empty optional if no bucket is specified.
     */
    public Optional<String> bucket() {
        return Optional.ofNullable(bucket);
    }
I updated to use a new S3BucketNotFoundException and created a test case.